presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java

```diff
@@ -76,5 +105,68 @@ public void testShowPartitionsFromHiveTable()
         String selectFromOnePartitionsSql = "show partitions from " + tableNameInDatabase;
         QueryResult partitionListResult = query(selectFromOnePartitionsSql);
         assertThat(partitionListResult).containsExactly(row(1), row(2));
+        assertColumnNames(partitionListResult, "part_col");
+    }
+
+    @Test(groups = {BASIC_SQL})
+    public void testShowPartitionsFromHiveTableWithTooManyPartitions()
+    {
+        String tableName = tablesState.get(PARTITIONED_TABLE_WITH_VARIABLE_PARTITIONS).getNameInDatabase();
+        createPartitions(tableName, TOO_MANY_PARTITIONS);
+
+        // Verify we created enough partitions for the test to be meaningful
+        assertExceptionThrownBy(() -> query("SELECT * FROM " + tableName))
+                .hasMessageMatching(".*: Query over table '\\S+' can potentially read more than \\d+ partitions");
+
+        QueryResult partitionListResult;
+
+        partitionListResult = query(format("SHOW PARTITIONS FROM %s WHERE part_col < 7", tableName));
+        assertThat(partitionListResult).containsExactly(row(0), row(1), row(2), row(3), row(4), row(5), row(6));
+        assertColumnNames(partitionListResult, "part_col");
+
+        partitionListResult = query(format("SHOW PARTITIONS FROM %s WHERE part_col < -10", tableName));
+        assertThat(partitionListResult).hasNoRows();
+    }
+
+    private void createPartitions(String tableName, int partitionsToCreate)
+    {
+        // This is done in tests rather than as a requirement, because TableRequirements.immutableTable cannot be partitioned
+        // and mutable table is recreated before every test (and this takes a lot of time).
+        // TODO convert to immutableTable + TableDefinition once immutableTable supports partitioned tables
+
+        requireNonNull(tableName, "tableName is null");
+
+        int maxPartitionsAtOnce = 100;
+
+        IntStream.range(0, IntMath.divide(partitionsToCreate, maxPartitionsAtOnce, RoundingMode.UP))
+                .forEach(batch -> {
+                    int rangeStart = batch * maxPartitionsAtOnce;
+                    int rangeEndInclusive = min((batch + 1) * maxPartitionsAtOnce, partitionsToCreate) - 1;
+                    query(format(
+                            "INSERT INTO %s (part_col, col) " +
+                                    "SELECT CAST(id AS integer), 42 FROM UNNEST (sequence(%s, %s)) AS u(id)",
+                            tableName,
+                            rangeStart,
+                            rangeEndInclusive));
+                });
+    }
+
+    private static AbstractThrowableAssert<?, ? extends Throwable> assertExceptionThrownBy(Runnable runnable)
```

Comment:
tempto provides such utility, please `com.teradata.tempto.assertions.QueryAssert#assertThat(com.teradata.tempto.assertions.QueryAssert.QueryCallback)`

Reply:
`com.teradata.tempto.assertions.QueryAssert.QueryExecutionAssert#failsWithMessage` 
1. accepts `String[]` for no reason 
2. does not accept pattern

Reply:
Why not to improve tempto. It should be easy.

Reply:
As a follow-up -- yes.

Reply:
Why is this needed? It seems to do the same as the AssertJ `assertThatThrownBy()` method